### ISSUE TITLE
tap is not a function

### ISSUE THREAD

**Issue by ivanovych666** [tgriesser/knex#3360]
# Environment

Knex version: 0.19.0

Example:
```
knex(tableName).select().tap().then(result => {
    console.log('success')
});
```

Result:
```
TypeError: knex(...).select(...).tap is not a function
```

Expected result:
```
success
```

**Comment by elhigu** [comment]
Knex is migrating out from bluebird. Native promises doesn't have that method.

**Comment by ivanovych666** [comment]
https://knexjs.org/#Interfaces-tap here is nothing about bluebird, here we can see that knex has .tap() method.

**Comment by ivanovych666** [comment]
> 0.19.0 is not released yet... so are you trying master?
Are you sure? https://www.npmjs.com/package/knex/v/0.19.0

**Comment by ricardograca** [comment]
`.tap()` comes from Blubird. Or rather, used to. The documentation just needs to be updated.

**Comment by ivanovych666** [comment]
This is a breaking change, all code where developers used this method won't work anymore. Why don't you won't leave this for backward compatibility?

**Comment by kibertoad** [comment]
@ivanovych666 It is, indeed, a breaking change, which is why it was included in 0.18.0 release, and communicated in migration guide (https://github.com/tgriesser/knex/blob/master/UPGRADING.md) -> `Knex returns native promises instead of bluebird ones now. You will need to update your code not to rely on bluebird-specific functionality;`. If you can't currently migrate your codebase to work with native promises, you should use 0.17.x

There was high demand from community to replace bluebird promises with native ones, and given the current direction of broader Javascript ecosystem, it definitely makes sense.

**Comment by kibertoad** [comment]
@elhigu 0.19.0 was released, actually. I will fix docs ASAP and hopefully release 0.19.1 tonight.

**Comment by kibertoad** [comment]
PR is up, please review!
